### ISSUE TITLE
How long to train a WAN or Hunyuan video lora?

### ISSUE THREAD

**Issue by MiloMindbender** [tdrussell/diffusion-pipe#131]
Can someone give me an idea of how long video training for Wan or Hunyuan takes on a 24gb 4090?  I'm just trying to get a rough idea of the time.  Just tell me how many videos/images and which model you trained on...then how long it took.

I'm just looking for rough numbers here, like if I want to train a WAN lora with 10 videos or 20 images is going to take minutes, hours or days on my local machine?

If anyone has training time comparisons between a local 4090 and a cloud machine with an 80gig GPU that would be great too.

I'm trying to decide if I want to use my local machine or a cloud one.  If local training times are long, I would probably use a cloud setup instead.

**Comment by GarbageHaus** [comment]
Hunyuan likeness LoRA on a 4090 with ~70 Epochs for a dataset around ~40 images and no repeats was about 1.25 hours.
Hunyuan motion/action LoRA on a 4090 with ~50 Epochs for a dataset around ~50 videos at 81 frames, 640x360 dimensions and no repeats was about 2.75 hours.

Training the base model will likely require multiple GPUs or a cloud machine, unless there is some technique I'm unaware of. Your experience may vary, I have only tried 1 action/motion LoRA and it didn't come out great. The likeness/image only LoRAs came out fine. I can't speak for Wan.

**Comment by MiloMindbender** [comment]
This is actually good news for me then.  With those times I can reasonably train stuff overnight, during lunch...etc.  Thanks!

I want to try some "likeness" Loras but am also interested in some motion stuff as I'm having trouble getting Wan or Hunyuan to do things like dancing or running reliably without it coming out spastic most of the time.

**Comment by Kokorop** [comment]
@GarbageHaus How do you get those timings? I'm training with only 20-30 photos on a 4090, and it takes about 5 or 6 hours to get a decent LoRA for a character (~50 epochs, ~1500 steps, and ~13 seconds per step, more or less). Maybe I should increase the learning rate? Right now, it's set to '2e-5', I don’t know if that’s a bit low, haha. Also, I usually do 4 to 5 repeats.

Maybe I need a good guide, I haven’t trained any LoRAs since early 2024, hahaha.
